trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestBuilder.java

```diff
@@ -98,7 +98,7 @@ record BucketAndKey(String bucket, String rawKey) {}
                 .flatMap(serverHostNameValue -> {
                     String lowercaseServerHostName = serverHostNameValue.toLowerCase(Locale.ROOT);
                     return Optional.of(request.requestUri().getHost())
-                            .filter(value -> value.endsWith(lowercaseServerHostName))
+                            .filter(value -> value.endsWith(lowercaseServerHostName) && !value.equals(lowercaseServerHostName))
```

Comment:
I believe that you can now remove the `endsWith(".")` check below right? You can simply make `serverHostNameValue` start with a `.`. That has two advantages:
- It would remove the need for you to check `!value.equals(lowercaseServerHostName)`
- You can also remove the check on line 103 to ensure the bucket doesn't end with a `.`